### ISSUE TITLE
PMIx_server_register_nspace fails since 3.1.3rc4

### ISSUE THREAD

**Issue by krempel-pt** [pmix/pmix#1355]
### What version of the PMIx Reference Library are you using?
v3.1.3rc4 and v3.1.3

### Describe how PMIx was installed
Compiled from the release candidate tarball using
configure --enable-debug CFLAGS=-DHAVE_SYSLOG

### Please describe the system on which you are running
Operating system/version: Fedora release 29
Computer hardware: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
Network type: none (local only)

-----------------------------

## Details of the problem

Since v3.1.3.rc4 the call of my host server to `PMIx_server_register_nspace()` results always in the `cbfunc` called with status `NOT-FOUND`.

Debugging output:
```
mca_base_verbose = level:9
pmix_server_base_verbose = 10
pmix_gds_base_verbose = 10
```
```
pmix[32465]: mca: base: component_find: searching (null) for gds components
pmix[32465]: mca: base: find_dyn_components: checking (null) for gds components
pmix[32465]: pmix:mca: base: components_register: registering framework gds components
pmix[32465]: pmix:mca: base: components_register: found loaded component ds12
pmix[32465]: pmix:mca: base: components_register: component ds12 has no register or open function
pmix[32465]: pmix:mca: base: components_register: found loaded component ds21
pmix[32465]: pmix:mca: base: components_register: component ds21 has no register or open function
pmix[32465]: pmix:mca: base: components_register: found loaded component hash
pmix[32465]: pmix:mca: base: components_register: component hash has no register or open function
pmix[32465]: mca: base: components_open: opening gds components
pmix[32465]: mca: base: components_open: found loaded component ds12
pmix[32465]: mca: base: components_open: component ds12 open function successful
pmix[32465]: mca: base: components_open: found loaded component ds21
pmix[32465]: mca: base: components_open: component ds21 open function successful
pmix[32465]: mca: base: components_open: found loaded component hash
pmix[32465]: mca: base: components_open: component hash open function successful
pmix[32465]: mca:gds:select: checking available component ds12
pmix[32465]: mca:gds:select: Querying component [ds12]
pmix[32465]: pmix:gds:dstore init
pmix[32465]: mca:gds:select: checking available component ds21
pmix[32465]: mca:gds:select: Querying component [ds21]
pmix[32465]: pmix:gds:dstore init
pmix[32465]: mca:gds:select: checking available component hash
pmix[32465]: mca:gds:select: Querying component [hash]
pmix[32465]: gds: hash init
pmix[32465]: Final gds priorities
pmix[32465]:         gds: ds21 Priority: 30
pmix[32465]:         gds: ds12 Priority: 20
pmix[32465]:         gds: hash Priority: 10
psid[32465]: PMIx Log Report:[32465]: pmix:server init called
pmix[32465]: [../../../../../src/mca/ptl/tcp/ptl_tcp_component.c:657] GDS STORE KV WITH hash
pmix[32465]: [pmix-server:32465] gds:hash:hash_store for proc [pmix-server:32465] key pmix.srvr.uri type PMIX_STRING scope STORE INTERNALLY
pmix[32465]: pmix_register_event_hdlr shifting to progress thread
pmix[32465]: pmix: register event_hdlr with 0 infos
pmix[32465]: pmix: _add_hdlr
pmix[32465]: pmix:server _register_nspace pspmix_0x00007ecc[0:32460]_32527
pmix[32465]: [../../src/server/pmix_server.c:582] GDS ADD NSPACE pspmix_0x00007ecc[0:32460]_32527
pmix[32465]: gds: dstore add nspace
pmix[32465]: ../../../../../src/mca/common/dstore/dstore_segment.c:169:pmix_common_dstor_create_new_segment: segment type 0, nspace pspmix_0x00007ecc[0:32460]_32527, id 0
pmix[32465]: ../../../../../src/mca/gds/ds21/gds_ds21_lock_pthread.c:151:pmix_gds_ds21_lock_init local_size 1
pmix[32465]: ../../../../../src/mca/common/dstore/dstore_segment.c:95:pmix_common_dstor_create_new_lock_seg: segment type 3, nspace pspmix_0x00007ecc[0:32460]_32527, id 0
pmix[32465]: gds: dstore add nspace
pmix[32465]: ../../../../../src/mca/common/dstore/dstore_segment.c:169:pmix_common_dstor_create_new_segment: segment type 0, nspace pspmix_0x00007ecc[0:32460]_32527, id 0
pmix[32465]: _../../../../../src/mca/gds/ds12/gds_ds12_lock_pthread.c:106:pmix_gds_ds12_lock_init_ _lockfile_name: /tmp/pmix_dstor_ds12_32465/dstore_sm.lock
pmix[32465]: [../../src/server/pmix_server.c:592] GDS CACHE JOB INFO WITH hash
pmix[32465]: [pmix-server:32465] gds:hash:cache_job_info for nspace pspmix_0x00007ecc[0:32460]_32527
pmix[32465]: pspmix: pspmix_server_registerNamespace: Callback from register namespace failed: NOT-FOUND
```

In the code I found this could indicate, that proc map or node map is not found.

But my info array contains both in an `PMIX_JOB_INFO_ARRAY` as the Standard v3.1 says:
```
info array to be passed to PMIx_server_register_nspace():
KEY: pmix.univ.size   PMIX_VALUE: Data type: PMIX_UINT32    Value: 1
KEY: pmix.ssn.arr     PMIX_VALUE: Data type: DATA_ARRAY     ARRAY SIZE: 2   { pmix.session.id(14)=0 - pmix.max.size(14)=1 }
KEY: pmix.job.size    PMIX_VALUE: Data type: PMIX_UINT32    Value: 1
KEY: pmix.job.arr     PMIX_VALUE: Data type: DATA_ARRAY     ARRAY SIZE: 5   { pmix.jobid(3)='pspmix_0x00000c79[0:3193]_3260' - pmix.job.size(14)=1 - pmix.max.size(14)=1 - pmix.nmap(3)='pmix[landroval]' - pmix.pmap(3)='pmix[0]' }
KEY: pmix.app.arr     PMIX_VALUE: Data type: DATA_ARRAY     ARRAY SIZE: 3   { pmix.appnum(14)='0' - pmix.app.size(14)=1 - pmix.aldr(40)=0 }
KEY: pmix.pdata       PMIX_VALUE: Data type: DATA_ARRAY     ARRAY SIZE: 8   { pmix.rank(40)=0 - pmix.lrank(13)=0 - pmix.nrank(13)=0 - pmix.nodeid(14)=0 - pmix.appnum(14)=0 - pmix.apprank(40)=0 - pmix.grank(40)=0 - pmix.spawned(1)=0 }
KEY: pmix.local.size  PMIX_VALUE: Data type: PMIX_UINT32    Value: 1
KEY: pmix.lpeers      PMIX_VALUE: Data type: PMIX_STRING    Value: 0
```

I do not understand what changed in between 3.1.3rc3 and 3.1.3rc4 that leads to the changed behavior and if this is intended or a bad side effect of something. Please help.

**Comment by rhc54** [comment]
Your code is fine - this is a problem on our side. My branch containing the implementation to support the info arrays (e.g., `PMIX_JOB_INFO_ARRAY`) has not yet been merged and missed the 3.1.3 release. However, the check for the two maps did get into the branch.

Bottom line: in 3.1.3rc3 we weren't properly parsing the data in the array (it just got stored as a single object). In 3.1.3 final release, we return an error because we don't recognize the maps.

I'll bring in the array support over the weekend and we will generate a 3.1.4 that fixes the problem.

**Comment by rhc54** [comment]
I fixed this in master with #1358 and it is included in the v3.1.x update #1359 . Thanks for the heads up!
